src/platforms/stm32/src/lib/sys.c

```diff
@@ -48,13 +51,13 @@ static void msleep(uint32_t delay)
 
 static inline void sys_clock_gettime(struct timespec *t)
 {
-    t->tv_sec = system_millis / 1000;
-    t->tv_nsec = (system_millis % 1000) * 1000000;
+    t->tv_sec = (time_t) system_millis / 1000;
+    t->tv_nsec = ((int32_t) system_millis % 1000) * 1000000;
 }
 
 static int32_t timespec_diff_to_ms(struct timespec *timespec1, struct timespec *timespec2)
 {
-    return (timespec1->tv_sec - timespec2->tv_sec) * 1000 + (timespec1->tv_nsec - timespec2->tv_nsec) / 1000000;
+    return (int32_t) ((timespec1->tv_sec - timespec2->tv_sec) * 1000 + (timespec1->tv_nsec - timespec2->tv_nsec) / 1000000);
```

Comment:
I'm not sure about this change, I mean what kind of warning are you fixing here?

Reply:
With no cast the specific error is:
```
AtomVM/src/platforms/stm32/src/lib/sys.c: In function 'timespec_diff_to_ms':
AtomVM/src/platforms/stm32/src/lib/sys.c:60:59: warning: conversion from 'time_t' {aka 'long long int'} to 'int32_t' {aka 'long int'} may change value [-Wconversion]
   60 |     return (timespec1->tv_sec - timespec2->tv_sec) * 1000 + (timespec1->tv_nsec - timespec2->tv_nsec) / 1000000;
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
The problem originates from adding **tv_sec** (defined as **time_t**, a.k.a. **long long int**) to **tv_nsec** (defined as **long int**).  It might make more sense to cast **tv_nsec** as `(time_t)` and have the function return a `time_t` rather than a `int32_t`, but upon closer examination this function is not used anywhere and may be dead code that can be removed.